core/templates/hash_map.h

```diff
@@ -129,11 +140,22 @@ class HashMap {
 	void _insert_with_hash(uint32_t p_hash, HashMapElement<TKey, TValue> *p_value) {
 		const uint32_t capacity = hash_table_size_primes[capacity_index];
 		const uint64_t capacity_inv = hash_table_size_primes_inv[capacity_index];
+		uint32_t pos = fastmod(p_hash, capacity_inv, capacity);
+
+		if (hashes[pos] == EMPTY_HASH) {
+			elements[pos] = p_value;
+			hashes[pos] = p_hash;
+
+			num_elements++;
+
```

Comment:
Just a small style nitpick: I'd remove the empty lines in this case. Both above and below num_elements++.